src/Symfony/Component/DependencyInjection/Tests/Loader/XmlFileLoaderTest.php

```diff
@@ -207,6 +208,35 @@ public function testParsesTags()
         }
     }
 
+    /**
+     * @expectedException \InvalidArgumentException
+     */
+    public function testParseTagsWithoutNameThrowsException()
+    {
+        try {
+            $container = new ContainerBuilder();
+            $loader = new XmlFileLoader($container, new FileLocator(self::$fixturesPath.'/xml'));
+            $loader->load('badtag2.xml');
+        } catch (InvalidArgumentException $e) {
+            throw $e->getPrevious();
+        }
+    }
+
+    /**
+     * @expectedException \InvalidArgumentException
+     * @expectedExceptionMessageRegExp /The tag name for service ".+" must not be an empty string/
+     */
+    public function testParseTagWithEmptyNameThrowsException()
+    {
+        try {
+            $container = new ContainerBuilder();
+            $loader = new XmlFileLoader($container, new FileLocator(self::$fixturesPath.'/xml'));
+            $loader->load('badtag5.xml');
+        } catch (InvalidArgumentException $e) {
+            throw $e->getPrevious();
```

Comment:
i think this is not executed at all. the exception you added above (which should be changed) is not wrapped at all.